### PR TITLE
Love: Entities card: add turn on/off all

### PR DESCRIPTION
**Pull request by c727**
fix: https://github.com/home-assistant/ui-schema/issues/35

also fixed some logic in the old ui

```yaml
    cards:
      - type: entities
        title: Toggle 
        entities:
          - input_boolean.switch_ac_kitchen
          - input_boolean.switch_ac_livingroom
          - input_boolean.switch_tv
          - light.bed_light
      - type: entities
        title: Toggle disabled
        show_header_toggle: false
        entities:
          - input_boolean.switch_ac_kitchen
          - input_boolean.switch_ac_livingroom
          - input_boolean.switch_tv
          - light.bed_light
      - type: entities
        title: Not toggleable
        entities:
          - binary_sensor.door
```

### CODE REVIEW

**Comment by c727** [comment]
correct STATES_ON in #1339

File: src/panels/lovelace/components/hui-entities-toggle.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        display: block;
+      }
+      paper-toggle-button {
+        cursor: pointer;
+        --paper-toggle-button-label-spacing: 0;
+        padding: 13px 5px;
+        margin: -4px -5px;
+      }
+    </style>
+    <template is="dom-if" if="[[_showToggle(_toggleEntities)]]">
```
Comment (balloob) [inline src/panels/lovelace/components/hui-entities-toggle.js]: `if="[[_toggleEntities.length]]"`

File: src/panels/lovelace/components/hui-entities-toggle.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    return entities.filter(entity => (entity in hass.states ?
+      canToggleState(hass, hass.states[entity]) : false));
+  }
+
+  _showToggle(entities) {
+    return entities.length !== 0;
+  }
+
+  _computeIsChecked(hass, entities) {
+    for (let i = 0; i < entities.length; i++) {
```
Comment (balloob) [inline src/panels/lovelace/components/hui-entities-toggle.js]:
```js
return entities.some(ent_id => STATES_ON.includes(hass.states[ent_id].state))
```

File: src/panels/lovelace/components/hui-entities-toggle.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            service = turnOn ? 'turn_on' : 'turn_off';
+            break;
+          default:
+            service = turnOn ? 'turn_on' : 'turn_off';
+        }
+
+        this.hass.callService(
+          serviceDomain,
+          service,
+          { entity_id: entity }
```
Comment (balloob) [inline src/panels/lovelace/components/hui-entities-toggle.js]: `entity_id` can be a list of entities. So we should group by domain/service.

```js
const toCall = {};
this.entities.forEach((ent_id) => {
  …
  key = [serviceDomain, service];
  if (!(key in toCall)) toCall[key] = [];
  toCall[key].push(ent_id)
});

Object.keys(toCall).forEach((key) => {
  const [domain, service] = key;
  const entities = toCall[key];
  this.hass.callService(…)
});
```

File: src/panels/lovelace/cards/hui-entities-card.js [hunk 4]
```diff
@@ -48,6 +51,7 @@ class HuiEntitiesCard extends EventsMixin(PolymerElement) {
     <ha-card>
       <div class='header'>
         <div class="name">[[_computeTitle(config)]]</div>
+        <hui-entities-toggle hass="[[hass]]" entities="[[config.entities]]"></hui-entities-toggle>
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-entities-card.js]: Feature request to turn this off incoming … 😉

Comment (c727) [reply src/panels/lovelace/cards/hui-entities-card.js]: what do you suggest as default?^^

Comment (balloob) [reply src/panels/lovelace/cards/hui-entities-card.js]: on.

Comment (balloob) [reply src/panels/lovelace/cards/hui-entities-card.js]: `show_header_toggle: false/true`

File: src/panels/lovelace/cards/hui-entities-card.js [hunk 5]
```diff
@@ -51,7 +51,9 @@ class HuiEntitiesCard extends EventsMixin(PolymerElement) {
     <ha-card>
       <div class='header'>
         <div class="name">[[_computeTitle(config)]]</div>
-        <hui-entities-toggle hass="[[hass]]" entities="[[config.entities]]"></hui-entities-toggle>
+        <template is="dom-if" if="[[_showHeaderToggle(config.show_header_toggle)]]">
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-entities-card.js]: No need for the function, just `if="[[config.show_header_toggle]]"`

**Review by balloob — APPROVED** [review]

**Pull request merged**